### ISSUE TITLE
What about .tsx extensions?

### ISSUE THREAD

**Issue by httpete** [yWorks/migrate-yfiles-app#15]
We always use tsx extensions, but it won't let me do that? This is a bug.

**Comment by yGuy** [comment]
TSX files are currently not supported, unfortunately. Some would argue that TSX is a bug, because it's not valid JavaScript or TypeScript ;-) , but from the perspective of the migration tool it's a missing feature or limitation, although a relatively important one for many.

That said,  to use the "analysis" part of the migration tool, it might work to check the output of tsc against the tool. If you are only migrating from the previous versions, chances are very high that the number of API changes is close to zero and getting a confirmation from the tool is enough, already. Modifying TSX code should be feasible if we use jscodeshift ( https://github.com/facebook/jscodeshift/issues/180 )

**Comment by httpete** [comment]
Yes Sebastian, it looks like TypeScript to the rescue, and the type support of yFiles is getting stronger in this release. We can get by - by hand for now since our yFiles footprint is small. But codemods would be great, like Angular ships between releases.
